app/models/application_choice.rb

```diff
@@ -1,4 +1,6 @@
 class ApplicationChoice < ApplicationRecord
+  before_create :set_vendor_id
+  self.primary_key = 'vendor_id'
```

Comment:
is it not enough to specify it on the migration?

Reply:
I can try, but from what I've seen primary keys specified in migrations assume the type is integer. :/